biosdk-services/src/main/java/io/mosip/biosdk/services/impl/spec_1_0/BioSdkServiceProviderImpl_V_1_0.java

```diff
@@ -134,6 +138,32 @@ public Object extractTemplate(RequestDto request) {
                     extractTemplateRequestDto.getModalitiesToExtract(),
                     extractTemplateRequestDto.getFlags()
             );
+            boolean isValid = true;
+            //Validation
+            if (response != null && response.getResponse() != null)
+            {
+        		BiometricRecord biometricRecord = (BiometricRecord)response.getResponse();
+                //Should have Valid BIR Segments
+        		List<BIR> segments = biometricRecord.getSegments();
+        		if (segments == null || segments.size() == 0)
+        		{
+        			isValid = false;
+        		}
+            }
+            else
+            {
+    			isValid = false;
+            }
+            if (!isValid)
+            {
+    	        logger.debug(LOGGER_SESSIONID, LOGGER_IDTYPE, "extractTemplate: Response Code  :: ", response.getStatusCode());        			
+    	        logger.debug(LOGGER_SESSIONID, LOGGER_IDTYPE, "extractTemplate: Response Status :: ", response.getStatusMessage());        			
+    	        logger.debug(LOGGER_SESSIONID, LOGGER_IDTYPE, "extractTemplate: Request :: ", request.getRequest());
+    	        logger.debug(LOGGER_SESSIONID, LOGGER_IDTYPE, "extractTemplate: Response :: ", response.getResponse());
+				response.setStatusCode(ResponseStatus.UNKNOWN_ERROR.getStatusCode());
+	            throw new BioSDKException(ErrorMessages.UNCHECKED_EXCEPTION.toString(), ErrorMessages.UNCHECKED_EXCEPTION.getMessage());
```

Comment:
When there such error happening within actual SDK, will it throw exception or it will just set the statusCode and statusMessage and respond back ? 
If it will just set and throw no exception, then we should also not through here, since we want just replicate what a typical SDK will do and not bring in any new behaviour.

Reply:
in the real sdk.. they should set statusmessage and statuscode.. should not throw error